### DISCUSSION TITLE
What's the best migration path from our stand-alone Wordpress site to a static site with code on Github?

### DISCUSSION THREAD

**Discussion by scottveirs** [orcasound/orcahome#123] [Ideas]
Since ~2002, we've used a stand-alone Wordpress site hosted by Dreamhost.com to provide general information about Orcasound. In 2018 we began building the Orcasound web app (in Elixir) and then in 2021 Isabella, one of the [Google Summer of Code students working with Orcasound](https://github.com/orcasound/orcagsoc/wiki), created this `orcahome` repository with the goal of migrating content there from the old Wordpress site.

Once the `orcahome` repo has reviewed and merged the existing pull requests submitted since Isabella finished her contributions...

**How shall we optimally conduct the migration from Wordpress to a static site generated via `orcahome` code?**

Sub-questions:

1. What are the best tools we could leverage to make the jump?
2. Is Netlify the best deployment solution, or are there new/better ideas that have emerged since Isabella and her GSoC mentor, Paul, decided on that route?
3. Where should content that is held in large files be stored (e.g. individual files Github's limit of 100 MB)?

Ideally, the process and tools will not breaking too much in the migration process (like SEO, i.e. permalinks that have been used to link to our content over the past couple decades). Perhaps folks can list other concerns or criteria that should be met, along with proposals in how to conduct the transition (e.g. in 2024)?

**Comment by paulcretu** [comment]
Reviving this thread, based on the discussion in today's call. Here are some thoughts on how to move the migration forward, at least incrementally:

- What if we create a new Dreamhost account and copy over the existing Wordpress instance? That way @UXBrendan, myself, and others can start playing with WP and see how far that gets us. If we're able to make the changes we need, whenever it's ready, we can switch the domain to point to the new instance.
  - Dreamhost has a [non-profit discount](https://help.dreamhost.com/hc/en-us/articles/215769478-Non-profit-discount). We'd need a 501(c)(3) to sponsor us (similar to the Slack discount). @scottveirs do you think we could ask one of the member orgs?
- We can try copying the WP [theme and config code](https://roots.io/bedrock/) into Github and set up [automated deploys](https://developer.wordpress.com/2021/08/11/automatically-deploy-to-wordpress-com-using-github-actions/). That way theme changes are accessible to open source contributions.
- Going even further, we could eventually set up a [static site plugin for WP](https://wordpress.org/plugins/simply-static/), since we only have static content. This way we could deploy to Github pages or some other static hosting provider. Though we would still need a hosted WP instance if we want a web UI for editing content, unless we're okay with using something like [Local](https://localwp.com/).

**Reply by UXBrendan** [reply]
Thanks for restarting this discussion and pulling some resources together. I like the approach of creating a new Dreamhost account. I've never looked at pricing, but if we can get non-profit sponsorship from a network partner, that would be ideal!

**Reply by paulcretu** [reply]
@adrmac If you're interested in making improvements to the WP site, this thread might be helpful
